### DISCUSSION TITLE
API Process - Site ID

### DISCUSSION THREAD

**Discussion by create-it-for-later** [amberelectric/public-api#235] [Q&A]
HI just looking for some assistance and guidance, as I seem to be stuck in setting up the API for our site. 

I can access the public data fine through an API, however when I add my Site ID and my authorisation psk token into the call, I get an unauthorised message response.  I did get an initial Site ID and psk token which I used for this.

When this failed, I dug deeper and found that that Site ID and NMI didn't match the NMI on my bill.
 Checking the amber site, I can see a Site ID and NMI. Note they are not the example ones and not my ones. 

If I go back into https://app.amber.com.au/developers and run the GET/Sites, which should return all our Site IDs. When I run it from the [app.amber.com](http://app.amber.com/) as a test the response I receive is
{
"message": "Unauthorised"}

Thank you 

Adam

**Comment by psmedley** [comment]
Have you clicked "Authorize' to add your api key before trying the same on the site?

The below should work to give you a list of sites - obvious replace psk_xxxxxxxxxxxxxx with your generated api key
`curl -X 'GET'   'https://api.amber.com.au/v1/sites'   -H 'accept: application/json'   -H 'Authorization: Bearer psk_xxxxxxxxxxxxxxxxx'
`

**Reply by create-it-for-later** [reply]
Thank you. 

It seems I tried to overcomplicate the process and missed the re-authorisation. 
It's always the way.
